packages/hooks/src/Provider.tsx

```diff
@@ -46,15 +48,50 @@ export const Provider: React.FC<ProviderProps> = ({ children, network, infuraId
   const [connected, setConnected] = React.useState<boolean>(false);
 
   const connectWallet = React.useCallback(async () => {
+    // Coinbase walletLink init
+    const walletLink = new WalletLink({
+      appName: 'coinbase',
+    });
+
+    // WalletLink provider
+    const walletLinkProvider = walletLink.makeWeb3Provider(
+      'https://eth-mainnet.alchemyapi.io/v2/qCdzfF9UqXcJYIle-Ff-BN0MII8LjLQs',
+      1
+    );
+
     const web3Modal = new Web3Modal({
       network,
       providerOptions: {
         walletconnect: {
           package: WalletConnectProvider,
           options: {
+            bridge: 'https://polygon.bridge.walletconnect.org',
             infuraId,
+            rpc: {
+              1: 'https://eth-mainnet.alchemyapi.io/v2/qCdzfF9UqXcJYIle-Ff-BN0MII8LjLQs', // mainnet // For more WalletConnect providers: https://docs.walletconnect.org/quick-start/dapps/web3-provider#required
```

Comment:
should we add more than just mainnet here, for the other networks that the user can pass in?

Reply:
I do think it could be done. I'll probably have to do some more research and find a way to do it.